typescript/processors/readTypeScriptModules.js

```diff
@@ -0,0 +1,451 @@
+var glob = require('glob');
+var path = require('canonical-path');
+var _ = require('lodash');
+var ts = require('typescript');
+
+module.exports = function readTypeScriptModules(tsParser, modules, getFileInfo, untouchedNamespaces,
+                                                getExportDocType, getContent, createDocMessage, log) {
+
```

Comment:
Hi @davidreher ,

I am new to dgeni, it might be an issue with my setup, but there is a thing I do not understand.

Statement from the [updated readme.md](https://github.com/davidreher/dgeni-packages/blob/add-typescript-package/README.md):

- we are not using a _**filereader**_

But the ```typescript package``` depends on the ```base package``` which gives me the following error if I am not using the filereader:

```sh
error:   Invalid property in "readFilesProcessor" (in "base" package)
error:    basePath=[Base path can't be blank], sourceFiles=[Source files can't be blank], fileReaders=[File readers can't be blank]
error:   Error processing docs:  processor=readFilesProcessor, package=base, basePath=[Base path can't be blank], sourceFiles=[Source files can't be blank], fileReaders=[File readers can't be blank]

/Users/ap/Sites/projects/MusicRightsReporting/Backend/node_modules/dgeni/node_modules/q/q.js:155
                throw e;
                ^
```

My setup looks like this:

```node
// Canonical path provides a consistent path (i.e. always forward slashes) across different OSes
var path = require('canonical-path');

var Package = require('dgeni').Package;

// Create and export a new Dgeni package called dgeni-example. This package depends upon
// the jsdoc and nunjucks packages defined in the dgeni-packages npm module.
module.exports = new Package('dgeni-example', [
  require('dgeni-packages/nunjucks'),
  require('dgeni-packages/typescript')
])
// this is required because the parsing-tags pseudo marker processor is defined in jsdoc package which is not referenced here.
  .processor({ name: 'parsing-tags', $runAfter: ['files-read'], $runBefore: ['processing-docs'] })

// Configure our dgeni-example package. We can ask the Dgeni dependency injector
// to provide us with access to services and processors that we wish to configure
  .config(function (log,
    readTypeScriptModules,
    templateFinder,
    writeFilesProcessor) {

    // Set logging level
    log.level = 'info';

    // Specify the base path used when resolving relative paths to source and output files
    readTypeScriptModules.basePath = path.resolve(__dirname, '..');

    // Specify collections of source files that should contain the documentation to extract
    readTypeScriptModules.sourceFiles = [
      {
        // Process all ts files in `src` and its subfolders ...
        include: 'src/**/*.ts',
        // ... except for this one!
        exclude: 'src/**/node_modules/**/*',
        // When calculating the relative path to these files use this as the base path.
        // So `src/foo/bar.ts` will have relative path of `foo/bar.ts`
        basePath: 'src'
      }
    ];

    // Add a folder to search for our own templates to use when rendering docs
    templateFinder.templateFolders.unshift(path.resolve(__dirname, 'templates'));

    // Specify how to match docs to templates.
    // In this case we just use the same static template for all docs
    templateFinder.templatePatterns.unshift('common.template.html');

    // Specify where the writeFilesProcessor will write our generated doc files
    writeFilesProcessor.outputFolder = 'dist/doc';
  });
```
It seems to be that the typescript processor should start after the read file processor (```$runAfter: ['files-read'],```), so I do not really get it..

What am I doing wrong? Do you have a simple example config?

Thanks,
Peter